homeassistant/components/buienradar/config_flow.py

```diff
@@ -90,27 +98,22 @@ async def async_step_user(
     ) -> FlowResult:
         """Handle a flow initialized by the user."""
         if user_input is not None:
-            lat = user_input.get(CONF_LATITUDE)
-            lon = user_input.get(CONF_LONGITUDE)
+            lat = user_input[CONF_LOCATION][CONF_LATITUDE]
```

Comment:
I'm not entirely sure but if I'm not mistaken location can be returned `None` if the user only press submit (using the found location).